File: ui2/src/model/file/FileOrUUID.js
import { BinaryFileDataset } from "./BinaryFileDataset";
export class FileOrUUID {
    fileid;
    binaryfiledataset;
    topublish;
    constructor(fileidOrFileOrUUID, binaryfiledataset, topublish) {
        if (this.fileid && typeof fileidOrFileOrUUID == "object") {
            Object.keys(this).forEach(
                (key) => (this[key] = fileidOrFileOrUUID[key] ? fileidOrFileOrUUID[key] : null)
            );
        } else {
            this.fileid = fileidOrFileOrUUID;
            this.binaryfiledataset = new BinaryFileDataset(binaryfiledataset);
            this.topublish = topublish;
        }
    }
}

File: src/main/java/fr/inra/oresing/rest/FileOrUUID.java
package fr.inra.oresing.rest;

import fr.inra.oresing.checker.DateLineChecker;
import fr.inra.oresing.model.Application;
import fr.inra.oresing.model.BinaryFileDataset;
import fr.inra.oresing.model.VariableComponentKey;
import fr.inra.oresing.persistence.DataRow;
import lombok.Getter;
import lombok.Setter;
import org.assertj.core.util.Strings;
import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
import org.springframework.util.CollectionUtils;
import org.testcontainers.shaded.org.apache.commons.lang.StringEscapeUtils;

import javax.annotation.Nullable;
import java.util.*;
import java.util.concurrent.atomic.AtomicInteger;
import java.util.stream.Collectors;

@Getter
@Setter
public class FileOrUUID {
    @Nullable
    UUID fileid;
    @Nullable
    BinaryFileDataset binaryfiledataset;
    Boolean topublish = false;

}
